### PR TITLE
KAFKA-16383: Ensure tasks have already polled their consumers before producing verified records in MirrorConnectorsIntegrationBaseTest::testReplicateFromLatest

### PR DESCRIPTION
**Pull request by C0urante**
[Jira](https://issues.apache.org/jira/browse/KAFKA-16383)

The flaky failures for this test appear to be due to a race condition. The current test steps are:
1. Produce 100 records to `test-topic-1`
2. Start the MM2 connectors
3. Wait for connector tasks to start (i.e., for there to be at least one task in the `RUNNING` state for each)
4. Produce 100 more records to `test-topic-1`

The assumption is that the `MirrorSourceTask` instance will poll its consumer in between steps 3 and 4 and seek to the end offset of `test-topic-1`. However, it's possible that the task instance will have completed startup and be marked `RUNNING` by the Kafka Connect runtime, but that it has not been polled yet. If this takes place and the task is polled (or rather, the task first polls its consumer) after step 4, then it will ignore those records when it seeks to the end of the topic.

As a fix, we wait for the task to replicate data from a "sentinel" topic first, before producing an additional 100 records to `test-topic-1`. This guarantees that the task has been able to poll its consumer and seek to the end of `test-topic-1`, and allows us to verify that the task should replicate exactly 100 records for that topic.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java [hunk 1]
```diff
@@ -1324,7 +1353,6 @@ private static <T> void waitForConsumingAllRecords(Consumer<T, T> consumer, int
     private static Map<String, String> basicMM2Config() {
         Map<String, String> mm2Props = new HashMap<>();
         mm2Props.put("clusters", PRIMARY_CLUSTER_ALIAS + ", " + BACKUP_CLUSTER_ALIAS);
-        mm2Props.put("max.tasks", "10");
```
Comment (C0urante) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java:1327]: Unrelated change: this property has no effect and should be removed so as not to cause FUD.

**Comment by C0urante** [comment]
@gharris1727 if you have time, would you mind reviewing this quick flaky test fix?

**Review by gharris1727 — APPROVED** [review]
